test/distributed/org/apache/cassandra/distributed/test/CqlConstraintsTest.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.distributed.test;
+
+import java.io.IOException;
+
+import org.junit.Test;
+
+import org.apache.cassandra.cql3.ConstraintInvalidException;
+import org.apache.cassandra.cql3.ConstraintViolationException;
+import org.apache.cassandra.distributed.Cluster;
+import org.apache.cassandra.distributed.api.ConsistencyLevel;
+import org.assertj.core.api.Assertions;
+import org.assertj.core.api.Condition;
+
+public class CqlConstraintsTest extends TestBaseImpl
```

Comment:
1. Add test cases for update. 
2. Add test cases for insert json. 
3. Add qt test cases that go over all applicable data types, i.e. numeric, blob, text, varchar, ascii, collections (which length should be applicable to).

Reply:
Talk offline. Collection is out of the scope for this patch. We do not need test case for this data type.